addons/web/static/src/js/control_panel/search_bar.js

```diff
@@ -239,16 +239,22 @@ odoo.define('web.SearchBar', function (require) {
             const { type } = this.props.fields[source.fieldName];
             const parser = field_utils.parse[type];
             let parsedValue;
-            if (['date', 'datetime'].includes(type)) {
-                const parsedDate = parser(rawValue, { type }, { timezone: true });
-                const dateFormat = type === 'datetime' ? 'YYYY-MM-DD HH:mm:ss' : 'YYYY-MM-DD';
-                const momentValue = moment(parsedDate, dateFormat);
-                if (!momentValue.isValid()) {
-                    throw new Error('Invalid date');
-                }
-                parsedValue = parsedDate.toJSON();
-            } else {
-                parsedValue = parser(rawValue);
+            switch (type) {
+                case 'date':
+                case 'datetime':
+                    const parsedDate = parser(rawValue, { type }, { timezone: true });
+                    const dateFormat = type === 'datetime' ? 'YYYY-MM-DD HH:mm:ss' : 'YYYY-MM-DD';
+                    const momentValue = moment(parsedDate, dateFormat);
```

Comment:
eslint: error
no-case-declarations - Unexpected lexical declaration in case block.

Reply:
Done